src/main/java/org/folio/edge/sip2/MainVerticle.java

```diff
@@ -54,7 +56,7 @@ public MainVerticle(Map<Command, ISip2RequestHandler> handlers) {
   }
 
   @Override
-  public void start() {
+  public void start(Future<Void> startFuture) {
```

Comment:
Since we are starting a server here, I think we should wait for the server to be up before we allow the verticle to become deployed. See: [Asynchronous Verticle start and stop](https://vertx.io/docs/vertx-core/java/#_asynchronous_verticle_start_and_stop).

Does that make sense?

Reply:
I remember we were talking about the verticle code. I think it was a misunderstanding and I probably convinced you to remove it, thought that was not my intention. 😄 I don't remember the details either. I just noticed it was missing and thought I'd put it back, since I remember it was there to begin with.